sqlglot/parser.py

```diff
@@ -625,7 +625,7 @@ def parse_into(self, expression_types, raw_tokens, sql=None):
                 return self._parse(parser, raw_tokens, sql)
             except ParseError as e:
                 error = e
-        raise ParseError(f"Failed to parse into {expression_types}") from error
+        raise ParseError.new(f"Failed to parse into {expression_types}") from error
```

Comment:
Ah, interesting.

This instance of `ParseError` will always have an empty `errors` list. It's really an aggregation of multiple sub errors.

I guess it won't immediately effect whatever you're doing, because you're probably not using `parse_into`.

But perhaps we need a different error type? Or maybe there's some clever way to concatenate all the sub errors together?

e.g.
```python
errors = []
for ...:
    try:
        ...
    except ParseError as e:
        errors.append(e)
raise ParseError(
    f"Failed to parse into {expression_types}", 
    errors=merge_errors(errors)
) from errors[-1]
```

That doesn't distinguish which errors are from which expression_types... but at least it keeps a more consistent ParseError interface.

Reply:
ah, yes, good point. Correct, I am not currently using `parse_into`, so I didn't think about capturing contextual errors here. I agree that being consistent in providing structures errors is better for the interface. 

I think your proposal in this comment is a good improvement, although I agree that there may be a way to simplify the concatenation further.

I've pushed and tested this approach, with the addition of an optional `into_expression` key per your comment about not distinguishing `expression_types`.

I think it could also be reasonable to make the `into_expression` key conditional, rather than often `None` for other types of `ParseErrors` but have no strong opinion either way.